lib/src/widgets/home_route_drawer.dart

```diff
@@ -0,0 +1,210 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class HomeRouteDrawer extends StatefulWidget {
+  const HomeRouteDrawer({super.key});
+
+  // Home Route Navigation Drawer State
+  @override
+  State<HomeRouteDrawer> createState() => _HomeRouteDrawerState();
+}
+
+class _HomeRouteDrawerState extends State<HomeRouteDrawer> {
+  final bool _isLoggedIn = false;
+  Future<void>? _browserLaunched;
+  int _homeRouteTile = 0;
+
+  Future<void> _launchBrowser(Uri url) async {
+    if (!await launchUrl(
+      url,
+      mode: LaunchMode.externalApplication,
+    )) {
+      throw Exception('Browser Launch Failed: $url');
+    }
+  }
+
+  Future<void> homeRouteTile(int index) async {
```

Comment:
What's the function of this?

Reply:
Navigation Drawer Tile. It was there previously before the Inherited Model was added and returned after it was removed.

Reply:
It's not a widget but a function for setting the `_homeRouteTile` index. However, I don't see the point of keeping this index at all since the menu is available only from the home route.

Reply:
If you remove that function, the Drawer will break, and you will get runtime errors. Try it.

https://m2.material.io/components/navigation-drawer/flutter#modal-navigation-drawer
https://api.flutter.dev/flutter/material/NavigationDrawer-class.html